provider/parameter.go

```diff
@@ -235,9 +242,14 @@ func parameterDataSource() *schema.Resource {
 							Description:  "The maximum of a number parameter.",
 							RequiredWith: []string{"validation.0.min"},
 						},
+						"monotonic": {
+							Type:        schema.TypeString,
+							Optional:    true,
+							Description: "Number monotonicity, either increasing or decreasing.",
```

Comment:
Just curious: Why don't we use a validation func here?

(I do see the check in `valueIsType`, it's just not obvious why we're doing that instead of validation func.)

Reply:
I will pass this question to @kylecarbs who originally implemented this code. I deduce that this is due to the compliance with `terraform-plugin-sdk`?

Reply:
Not sure actually...